### PR TITLE
[ENHANCEMENT] Condense column-level `vconcat` plots into one interactive plot

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- GREAT-812
- Condense column domain charts into one visualization

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py [hunk 1]
```diff
@@ -171,23 +191,28 @@ def get_expect_values_to_be_between_chart(
             metric_type: The altair data type for the metric being plotted
             domain_name: The name of the domain as it exists in the pandas dataframe
             domain_type: The altair data type for the domain being plotted
+            subtitle: The subtitle to add for a domain such as "Column: column_name"
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py:194]: 🙇🏽 thank you for this

File: great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py [hunk 2]
```diff
@@ -274,7 +290,505 @@ def _determine_anomaly_coded_line(
...
+                & (alt.datum.max_value >= alt.datum[metric_name])
+            ) | (
+                (alt.datum.min_value <= alt.datum[metric_name])
+                & (alt.datum.max_value <= alt.datum[metric_name])
+            )
+        point_color_condition: alt.condition = alt.condition(
+            predicate=predicate,
+            if_false=alt.value(Colors.GREEN.value),
+            if_true=alt.value(Colors.PINK.value),
+        )
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py:766]: Could some of these longer functions be refactored into helpers? I know Altair is quite verbose by default but I think there's room for some increment refactoring (perhaps after this PR).

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py [hunk 3]
```diff
@@ -41,29 +41,38 @@ def plot(
...
+            column_domain_return_charts,
+        ) = self._plot_column_domain_charts(
             expectation_configurations=expectation_configurations,
             include_column_names=include_column_names,
             exclude_column_names=exclude_column_names,
             prescriptive=prescriptive,
         )
-        charts.extend(column_domain_chart)
+        display_charts.extend([column_domain_display_charts])
+        return_charts.extend([column_domain_return_charts])
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:71]: We'd need a conditional to ensure that we use the base display for single column tables right?

Comment (cdkini) [reply great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:71]: Also is there a reason we're putting these in lists? Aren't they already lists?

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py [hunk 4]
```diff
@@ -165,7 +182,7 @@ def _create_chart_for_table_domain_expectation(
             prescriptive=prescriptive,
         )
 
-        return self._chart_values(
+        return self._chart_domain_values(
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:185]: Appreciate the rename 😄

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py [hunk 5]
```diff
@@ -41,29 +41,38 @@ def plot(
                 "You may either use `include_column_names` or `exclude_column_names` (but not both)."
             )
 
-        charts: List[alt.Chart] = []
+        display_charts: List[alt.Chart] = []
+        return_charts: List[alt.Chart] = []
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:45]: Could we update the docstring to reflect the difference between these two?

**Review by cdkini — APPROVED** [review]
We might need some minor refactors but as this is functional, I'm more than happy to approve ✅

File: great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py (hunk 5 above)
Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py:45]: ✅

**Pull request merged**